.github/workflows/docs.yml

```
@@ -16,13 +17,29 @@ jobs:
     permissions:
       contents: write
       pull-requests: write
+    env:
+      BIN_DIR: ${{ github.workspace }}/bin
+      DOXYGEN_URL: 'https://github.com/doxygen/doxygen/releases/download/Release_1_10_0/doxygen-1.10.0.linux.bin.tar.gz'
+      DOXYGEN_VERSION: 1.10.0
     steps:
       - uses: actions/checkout@v4
       - uses: ./.github/actions/setup
 
       - name: Install dependencies
         run: |
-          sudo apt-get install -y doxygen python3-msgpack
+          sudo apt-get install -y python3-msgpack
+
+      - name: Install doxygen
+        run: |
+          curl --retry 5 --silent --show-error --fail -L -o /tmp/doxygen.tar.gz "$DOXYGEN_URL"
+          mkdir -p "$BIN_DIR" /opt/doxygen
+          tar xfz /tmp/doxygen.tar.gz --strip-components=1 -C /opt/doxygen
+          ln -sfn /opt/doxygen/bin/doxygen "$BIN_DIR/doxygen"
+          doxygen_version="$(doxygen -v)"
+          echo "$doxygen_version" | grep -qF "$DOXYGEN_VERSION" || {
+            echo "Unexpected doxygen version: $doxygen_version";
+            exit 1;
+          }
```

Humor me, can you test this and check that it doesn't take too long? I'm wondering if the extra CI time is worth the simplicity.

```suggestion
- uses: cachix/install-nix-action@v25
- run: nix-env -iA doxygen -f https://github.com/NixOS/nixpkgs/archive/9957cd48326fe8dbd52fdc50dd2502307f188b0d.tar.gz
```

---

Actually not too bad: https://github.com/wookayin/neovim/actions/runs/7856673696/job/21439682545?pr=3

took 4s for installing nix and 19s for installing doxygen via nix. The baseline (tarball) is 2s. So the additional cost is around 20 seconds.

---

Eh, unsure if it's worth it. Up to you. Anyway, thanks for testing it.